process/headerCheck/headerSignatureVerify.go

```diff
@@ -290,25 +294,43 @@ func (hsv *HeaderSigVerifier) VerifySignatureForHash(header data.HeaderHandler,
 	return multiSigVerifier.VerifyAggregatedSig(pubKeysSigners, hash, signature)
 }
 
-// VerifyHeaderWithProof checks if the proof on the header is correct
-func (hsv *HeaderSigVerifier) VerifyHeaderWithProof(header data.HeaderHandler) error {
+// VerifyHeaderWithPrevProof checks if the proof on the header is correct
+func (hsv *HeaderSigVerifier) VerifyHeaderWithPrevProof(header data.HeaderHandler) error {
 	// first block for transition to equivalent proofs consensus does not have a previous proof
-	if !common.ShouldBlockHavePrevProof(header, hsv.enableEpochsHandler, common.EquivalentMessagesFlag) {
-		if prevProof := header.GetPreviousProof(); !check.IfNilReflect(prevProof) {
-			return ErrProofNotExpected
-		}
-		return nil
-	}
-
 	err := verifyPrevProofForHeaderIntegrity(header)
 	if err != nil {
 		return err
 	}
 
 	prevProof := header.GetPreviousProof()
+	if common.IsEpochChangeBlockForFlagActivation(header, hsv.enableEpochsHandler, common.EquivalentMessagesFlag) {
+		return hsv.verifyHeaderProofAtTransition(header, prevProof)
```

Comment:
I don't think this would ever be called, as this method, `VerifyHeaderWithPrevProof`, is called only if `ShouldBlockHavePrevProof`, but if block should have proof, it means it is not the epoch change block for flag activation, thus `VerifySignature` would go through L238-L250 for the epoch start block